[crmlatte] iP

crmlatte opened the pull request:
### Sebastian

> "I can accept failure. Everyone fails at something. But I can't accept not trying." - Micheal Jordan

Sebastian is your _friend_ and remembers your tasks for you so that you don't have to!
Face no failures as all your brain space is for thinking, not holding tasks

Sebastian is: 
- easy to use
- helpful
- super fast 😁😁

All you need to do is,
1. download it from [here](https://github.com/nus-cs2103-AY2425S2/ip/pull/124)
2. double-click it
3. add your tasks
4. let him manage your tasks for you 😉

And it is **FREE**!

Features:
- [ ]  Managing tasks
- [x]  Managing deadlines (coming soon)
- [x]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
````java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
change name, greet, and exit

Commit:
echo

Commit:
add, list

Commit:
mark as done/undone

Commit:
todo, event, deadline

Commit:
automated text ui testing

Commit:
Handle errors

Commit:
delete tasks

Commit:
adding storage to save list to device

Commit:
formatting

Commit:
spacing

Commit:
edit deadline class

Commit:
Revert "edit deadline class"

This reverts commit 8c55d93a02f36f9a3d9fa51d40d37968064624b2.

Commit:
Merge branch 'master' into branch-Level-7

Commit:
create class Ui and Parser

Commit:
test branch

Commit:
ooppp

Commit:
Ui package

Commit:
using date time object instead of string

Commit:
add "date" function to return all tasks due on input date

Commit:
spacing

Commit:
Revert "spacing"

This reverts commit 0bd137d845e7516c20e9557aef73dab63f009cec.

Commit:
Merge branch 'master' into branch-Level-8

* master:
  Revert "spacing"
  spacing

Commit:
Merge commit '81a9c537afe8492675eb253968939364d65db091'

* commit '81a9c537afe8492675eb253968939364d65db091':
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
Merge branch 'master' into branch-Level-8

* master:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

Commit:
add "find" function that prints matching tasks to given keyword

Commit:
Changes according to java coding standard

Commit:
Add javadoc for all non-private methods

Commit:
Add comments

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Changes according to java coding standard

Commit:
Merge branch 'master' into branch-A-JavaDoc

* master:
  Add comments
  Changes according to java coding standard

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  add "find" function that prints matching tasks to given keyword

# Conflicts:
#	src/main/java/seb/ui/Sebastian.java

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add javadoc for all non-private methods

Commit:
Merge branch 'branch-Level-8'

* branch-Level-8:
  add "date" function to return all tasks due on input date
  using date time object instead of string

# Conflicts:
#	src/main/java/seb/ui/Sebastian.java
#	src/main/java/seb/ui/TaskList.java
#	src/main/java/seb/ui/Ui.java

Commit:
variable changes

Commit:
Add javadoc for parseDateTime() and parseShowDate() in Parser class

Commit:
Error handling in showDates() method for no tasks on given date
returns different output message

Commit:
Add javadoc for find() method in Ui class

Commit:
Formatting of output

Commit:
TaskList class: move showDates() method from
TaskList to Ui class
Ui class: edit showDates() to show number of tasks due
Sebastian class: edit "DATE" switch case

Commit:
test(TaskList): Add unit tests for TaskList methods

- Added JUnit tests for adding and removing tasks in TaskList
- Ensured tasks are correctly stored and removed from the list
- Used @BeforeEach to initialize a fresh TaskList instance before each test

These tests improve coverage and ensure TaskList behaves as expected

Commit:
fix(Parser): Correct trimming logic in parseEvent method
test(Parser): Add unit tests for parsing methods

- Fixed the for loop in parseEvent to properly trim and store values
- Ensured correct splitting logic for event input format
- Added JUnit tests in ParserTest to verify parseTodo, parseDeadline, and parseEvent behavior
- Improved error handling for invalid event formats

This improves input parsing reliability and increases test coverage

Commit:
Change jar file name in build.gradle

- Changed to seb.ui.Sebastian, where the main() method is found

Commit:
chore(checkstyle): Add Checkstyle configuration files

- Added `checkstyle.xml` and `suppressions.xml` in `config/checkstyle`
- Configured Checkstyle to enforce coding standards in the project
- Ensured suppression of specific rules as needed

This improves code consistency and maintainability.

Commit:
style(checkstyle): Fix Checkstyle violations in codebase

- Updated code to comply with Checkstyle rules
- Fixed indentation, naming conventions, and formatting issues
- Ensured consistent coding style across the project

This improves code readability and maintainability.

Commit:
Merge tag 'A-CheckStyle'

* tag 'A-CheckStyle':
  chore(checkstyle): Add Checkstyle configuration files

zephaniahong commented on src/main/java/seb/ui/Deadline.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public String toFileFormat() {
```
This could potentially be misleading as it sounds as if you are converting it to a particular file format e.g. csv, txt etc...

Commit:
feat(gui): Add initial setup for GUI implementation

- Added `Main`, `Launcher`, and `DialogBox` classes for GUI integration
- Updated `build.gradle` to support JavaFX dependencies
- Added necessary images in `main/resources/images`
- Prepared the project structure for future GUI enhancements

This sets up the foundation for the graphical user interface.

zephaniahong commented on src/main/java/seb/ui/Parser.java:
```diff
@@ -0,0 +1,127 @@
+     * @param input Args of Command object
+     * @return integer value of String number
+     */
+    public static int parseNum(String input) {
+        return Integer.parseInt(input);
+    }
```
Is there a reason for creating a wrapper function over the library function?

zephaniahong commented on src/main/java/seb/ui/Parser.java:
```diff
@@ -0,0 +1,127 @@
+            DateTimeFormatter formatInput = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+            LocalDateTime dateTime = LocalDateTime.parse(input, formatInput);
+            DateTimeFormatter formatOutput = DateTimeFormatter.ofPattern("E, dd MMM yyyy HH:mm");
+            String formattedDT = dateTime.format(formatOutput);
+            return formattedDT;
```
```suggestion
            String formattedDt = dateTime.format(formatOutput);
            return formattedDt;
```

zephaniahong commented on src/main/java/seb/ui/Parser.java:
```diff
@@ -0,0 +1,127 @@
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+            LocalDate dateTime = LocalDate.parse(input, formatter);
+            DateTimeFormatter formatOutput = DateTimeFormatter.ofPattern("E, dd MMM yyyy");
+            String ftd = dateTime.format(formatOutput);
```
Consider using a more descriptive variable name

zephaniahong commented on src/main/java/seb/ui/Ui.java:
```diff
@@ -0,0 +1,104 @@
+     * Prints all tasks stored in ArrayList
+     * @param tasks list of all tasks
+     */
+    public void list(ArrayList<Task> tasks) {
```
Maybe you could consider naming this method `listTasks`?

zephaniahong commented on src/main/java/seb/ui/Ui.java:
```diff
@@ -0,0 +1,104 @@
+     * @param tasks ArrayList tasks
+     * @param search String of keyword to match
+     */
+    public void find(ArrayList<Task> tasks, String search) {
```
Consider giving this method a more descriptive name

zephaniahong reviewed (commented):
Great job overall! Just a couple of small nits

Commit:
feat(gui): Integrate FXML for MainWindow and DialogBox

- Added `MainWindow.fxml` and `DialogBox.fxml` for GUI layout
- Updated `Main`, `Launcher`, and `DialogBox` classes to load FXML files
- Improved UI structure for better separation of logic and presentation

This enhances the GUI by using FXML for better maintainability and scalability

Commit:
feat(gui): Integrate FXML for MainWindow and DialogBox

- Added `MainWindow.fxml` and `DialogBox.fxml` for GUI layout
- Updated `Main`, `Launcher`, and `DialogBox` classes to load FXML files
- Improved UI structure for better separation of logic and presentation

This enhances the GUI by using FXML for better maintainability and scalability

Commit:
refactor(Sebastian): Capture Ui output for GUI integration

- Added `captureOutput()` method to redirect and capture `Ui` class prints
- Updated `getResponse()` to use `Ui` methods without modifying them
- Ensured `Sebastian` can return responses for both CLI and GUI
- Preserved CLI functionality while enabling GUI interaction

This allows seamless integration of `Sebastian` with the GUI without altering `Ui` methods

Commit:
refactor(Sebastian): Capture Ui output for GUI integration

- Added `captureOutput()` method to redirect and capture `Ui` class prints
- Updated `getResponse()` to use `Ui` methods without modifying them
- Ensured `Sebastian` can return responses for both CLI and GUI
- Preserved CLI functionality while enabling GUI interaction

This allows seamless integration of `Sebastian` with the GUI without altering `Ui` methods

Commit:
style(ui): Update Ui messages for more personality

- Enhanced messages for a friendlier tone
- Refined task-related responses to be more engaging
- Improved overall conversational feel of the assistant

These changes make interactions with Sebastian more lively and user-friendly

Commit:
style(ui): Update Ui messages for more personality

- Enhanced messages for a friendlier tone
- Refined task-related responses to be more engaging
- Improved overall conversational feel of the assistant

These changes make interactions with Sebastian more lively and user-friendly

Commit:
Merge tag 'Level-10'

* tag 'Level-10':
  feat(gui): Add initial setup for GUI implementation

Commit:
Merge tag 'Level-10'

* tag 'Level-10':
  refactor(Sebastian): Capture Ui output for GUI integration
  feat(gui): Integrate FXML for MainWindow and DialogBox

Commit:
Merge branch 'master' into Level-10

* master:
  refactor(Sebastian): Capture Ui output for GUI integration
  feat(gui): Integrate FXML for MainWindow and DialogBox

Commit:
Merge branch 'master' into Level-10

* master:
  style(ui): Update Ui messages for more personality

Commit:
Merge commit '41853c8f9759de94a49c1625f0c16b69edfd0543'

* commit '41853c8f9759de94a49c1625f0c16b69edfd0543':
  style(ui): Update Ui messages for more personality

Commit:
Merge commit '6767bc5e9ed8d1c647be2719042b5c1d347ea462'

* commit '6767bc5e9ed8d1c647be2719042b5c1d347ea462':
  refactor(Sebastian): Capture Ui output for GUI integration
  feat(gui): Integrate FXML for MainWindow and DialogBox

Commit:
Merge branch 'master' into Level-10

* master:

Commit:
feat(gui): Add Sebastian's photo to welcome message

- Updated `MainWindow` to display Sebastian's welcome message with his image on startup
- Ensured `DialogBox` properly handles the welcome message format
- Loaded `sebphoto.jpeg` from resources to maintain consistency with response messages

This improves the user experience by making the GUI more engaging

Commit:
refactor: add assertions to document critical assumptions

Assertions have been added to key areas of the codebase to document
assumptions that should always hold. This helps catch logic errors
early during development and improves code reliability

Commit:
refactor(storage): Improve code quality across classes

- Extracted task parsing logic into a new parseTaskLine method
  to improve readability and maintainability.
- Updated Javadoc comments for better clarity and consistency.
- Added input validation in parseTaskLine to prevent
  malformed task entries from causing runtime errors.

This refactoring makes loadTasks more modular and easier to
extend in the future while maintaining robust error handling.

crmlatte committed:
Merge pull request #2 from crmlatte/branch-A-Assertions

refactor: add assertions to document critical assumptions

Commit:
Merge branch 'branch-A-CodeQuality'

* branch-A-CodeQuality:
  refactor(storage): Improve code quality across classes

crmlatte committed:
Merge pull request #3 from crmlatte/branch-A-CodeQuality

refactor(storage): Improve code quality across classes

Commit:
Merge branch 'master' of https://github.com/crmlatte/ip

* 'master' of https://github.com/crmlatte/ip:

Commit:
fix(parser): Correct date-time parsing for Event and Deadline

- Updated Parser logic to properly parse date-time inputs into
  the correct LocalDateTime format.
- Ensured consistent date-time formatting when creating new Event
  and Deadline tasks.

This fixes inconsistencies in date-time parsing, improving
task creation accuracy and user experience.

Commit:
feat: Add update command to edit existing tasks

- Implemented UPDATE command in Sebastian to allow users to modify
  the description of existing tasks.
- Prompted users for a second input to enter the new task description.
- Added update method to Task to support updates.
- Ensured tasks are updated and saved correctly in Storage.

This enhances task management by allowing users to modify tasks
without deleting and recreating them.

Commit:
feat(parser): add date-time validation in checkDateValidity

- Implemented checkDateValidity method in Parser class.
- Ensures that dates are not in the past.
- Ensures that the start date is not after the end date.

Commit:
test: add TaskTest for update method and expand ParserTest for checkDateValidity

- Added TaskTest class to test the update method in Task class.
- Covered cases for updating description, valid deadlines, past deadlines, and invalid fields.
- Added more test cases in ParserTest to validate checkDateValidity method.
- Ensured robust exception handling and date validation.

Commit:
feat: enhance Sebastian personality and add fun responses with
sebby method

- Edited Sebastian responses to be more engaging and expressive.
- Added sebby method to generate fun and playful responses
for better user experience.
- Improved interaction flow to make conversations feel more
natural and enjoyable.

Commit:
feat(gui): set window title and disable input after "bye"

- Added window title "Sebastian" in the GUI.
- Disabled user input after the "bye" command is given.
- Ensured smooth user experience by preventing further input.

Commit:
fix(parser): allow events with past start, future end

- Updated `checkDateValidity` to permit events where
  the start date is in the past but the end date is
  in the future.
- Ensured such events can still be updated correctly.
- Maintained validation for other invalid dates.

Commit:
fix(parser): update parseUpdate to handle CLI input

- Adjusted `parseUpdate` logic to correctly process command
  line input for updating tasks.
- Ensured tasks are updated as expected based on user input.
- Improved handling of edge cases for better reliability.

Commit:
feat(gui): enhance UI with new font, colors, and spacing

- Updated font style to improve readability and aesthetics.
- Adjusted background and colours for a more engaging experience.
- Improved spacing to enhance layout and user interaction.
- Created a more fun and visually appealing UI.

Commit:
chore(docs): add app screenshot to docs folder

Commit:
docs: update README with latest features

- Updated README to reflect all features of Sebastian
- Added more details for better clarity and usability.
- Improved formatting and structure for readability.

Commit:
docs: add additional screenshots

crmlatte committed:
Update README.md

crmlatte committed:
Merge pull request #5 from crmlatte/crmlatte-patch-1

Update README.md
